src/oas2/transformers/params.ts

```diff
@@ -81,41 +82,84 @@ export function translateToBodyParameter(body: BodyParameter, consumes: string[]
 }
 
 export function translateFromFormDataParameter(
-  form: FormDataParameter,
-  requestBody: IHttpOperationRequestBody | null | undefined,
+  parameters: FormDataParameter[],
   consumes: string[],
 ): IHttpOperationRequestBody {
-  const baseContent = buildContentFromParameter(form);
-
-  const requestBodyCopy: IHttpOperationRequestBody = Object.assign(
-    {},
-    requestBody || {
-      allowEmptyValue: form.allowEmptyValue,
-      contents: consumes.map(mediaType => ({
-        ...baseContent,
-        mediaType,
-      })),
-    },
-  );
-  const bodyContent = requestBodyCopy.contents && requestBodyCopy.contents[0];
-
-  if (bodyContent) {
-    bodyContent.schema = bodyContent.schema || baseContent.schema;
+  const finalBody: IHttpOperationRequestBody = {
+    contents: consumes.map(mediaType => ({
+      mediaType,
+      schema: {
+        type: 'object',
+      },
+    })),
+  };
 
-    const encoding: IHttpEncoding = {
-      property: form.name,
-      style: HttpParamStyles.Form,
-    };
+  return parameters.reduce((body, parameter) => {
+    const { schema } = buildSchemaForParameter(parameter);
+    (body.contents || []).forEach(content => {
+      // workaround... JSONSchema4 does not support `allowEmptyValue`
+      if ('allowEmptyValue' in parameter) {
+        // @ts-ignore
+        schema.allowEmptyValue = parameter.allowEmptyValue;
```

Comment:
You can use `Object.assign` to avoid to shutdown the type checker on this line

```suggestion
        Object.assign(schema, { allowEmptyValue: parameter.allowEmptyValue });
```